src/Ads/AssetSuggestionsService.php

```diff
@@ -17,13 +19,181 @@
+	 * @throws Exception If the Post ID is invalid.
+	 */
+	protected function get_post_assets( int $id ): array {
+		$post = $this->wp->get_post( $id );
+
+		if ( ! $post ) {
+			throw new Exception(
+				/* translators: 1: is a integer representing an unknown Post ID */
+				sprintf( __( 'Invalid Post ID %1$d', 'google-listings-and-ads' ), $id )
+			);
+		}
+
+		$attachments_ids = $this->get_post_attachments(
+			[
+				'post_parent' => $id,
+			]
+		);
+
+		if ( $id === wc_get_page_id( 'shop' ) ) {
+			$attachments_ids = array_merge( $attachments_ids, $this->get_shop_attachments() );
+		}
+
+		if ( $post->post_type === 'product' || $post->post_type === 'product_variation' ) {
+			$product         = $this->wc->maybe_get_product( $id );
+			$attachments_ids = array_merge( $attachments_ids, $product->get_gallery_image_ids(), [ $product->get_image_id() ] );
+		}
+
+		$gallery_images_urls = get_post_gallery_images( $id );
+		$marketing_images    = array_merge( $this->get_url_attachments_by_ids( $attachments_ids ), $gallery_images_urls );
+		$marketing_images    = array_slice( $marketing_images, 0, self::DEFAULT_MAXIMUM_MARKETING_IMAGES );
+		$long_headline       = get_bloginfo( 'name' ) . ': ' . $post->post_title;
+
+		return [
+			'headline'                => [ $post->post_title ],
+			'long_headline'           => [ $long_headline ],
+			'description'             => $this->remove_empty_values( [ $post->post_excerpt, get_bloginfo( 'description' ) ] ),
+			'logo'                    => $this->remove_empty_values( [ wp_get_attachment_image_url( get_theme_mod( 'custom_logo' ) ) ] ),
+			'final_url'               => get_permalink( $id ),
+			'business_name'           => get_bloginfo( 'name' ),
+			'display_url_path'        => [ $post->post_name ],
+			'square_marketing_images' => $marketing_images,
+			'marketing_images'        => $marketing_images,
+			'call_to_action'          => null,
+		];
+	}
+
+	/**
+	 * Get attachments related to the shop page.
+	 *
+	 * @return array Shop attachments.
+	 */
+	protected function get_shop_attachments(): array {
+		return $this->get_post_attachments(
+			[
+				'post_parent__in' => $this->get_shop_products(),
+			]
+		);
+	}
+
+	/**
+	 *
+	 * Get products that will be use to offer image assets.
+	 *
+	 * @param array $args See WP_Query::parse_query() for all available arguments.
+	 * @return array Shop products.
+	 */
+	protected function get_shop_products( array $args = [] ): array {
+		$defaults = [
+			'post_type'   => 'product',
+			'numberposts' => self::DEFAULT_MAXIMUM_MARKETING_IMAGES,
+			'fields'      => 'ids',
+		];
+
+		$args = wp_parse_args( $args, $defaults );
+
+		return $this->wp->get_posts( $args );
+	}
+
+	/**
+	 * Remove empty values from array.
+	 *
+	 * @param array $array A list of strings.
+	 *
+	 * @return array A list of strings without empty strings.
+	 */
+	protected function remove_empty_values( array $array ): array {
+		return array_values( array_filter( $array ) );
+	}
```

Comment:
💅 This is more like a utility method so might be better to create something that is shareable like `ArrayUtil.php` or a trait. But it's out of the scope of this PR so we can do it maybe in the cooldown for some refactoring.